api/v1alpha1/openapi.yaml

```diff
@@ -2181,6 +2181,141 @@ components:
         - "ResourceAlertSeverityTypeWarning"
         - "ResourceAlertSeverityTypeCritical"
         - "ResourceAlertSeverityTypeInfo"
+    DeviceConfigHookSpec:
+      type: object
+      properties:
+        name:
+          type: string
+        description:
+          type: string 
+        actions:
+          type: array
+          items:
+            $ref: '#/components/schemas/ConfigHookAction'
+        watchPath:
+          type: string
+          description: 'The path to monitor for changes in configuration files. This path can point to either a specific file or an entire directory.'
+      required:
+        - name
+        - description
+        - actions
+        - watchPath
+    FileOperation:
+      type: string
+      description: 'The type of operation that was observed on the file.'
+      enum:
+        - "Create"
+        - "Remove"
+        - "Update"
+        - "Delete"
+        - "Rename"
+        - "ChangePermissions"
```

Comment:
Similar to the above comment - will the agent change permissions or delete+create?

Reply:
Makes sense

Reply:
fixed